store/sqlstore/store.go

```diff
@@ -1167,6 +1172,24 @@ func (ss *SqlStore) ensureMinimumDBVersion(ver string) (bool, error) {
 	return true, nil
 }
 
+func (ss *SqlStore) ensureDatabaseCollation() error {
+	if *ss.settings.DriverName == model.DatabaseDriverMysql {
+		var dbCollation struct {
+			Variable_name string
+			Value         string
+		}
+		if err := ss.GetMasterX().Get(&dbCollation, "SHOW VARIABLES LIKE 'collation_database'"); err != nil {
```

Comment:
This can be slightly tricky with permissions. Some setups don't give full permissions to the mattermost user to run all system level commands. In that case, it might fail. Wondering if we can detect that and ignore if we see such an error?

Reply:
I couldn't find anything restring a user to read global variables. I tried with a non-privileged user (has only SELECT privilege to a table) and it can still read the global variable. Do you know if it's possible to restrict users from reading variables?

Reply:
Ah sounds good then. I thought this was a privileged operation.